src/auth/webhook-secret.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import * as crypto from 'crypto';
+import { Request } from 'express';
+import { Env } from 'src/env';
+
+@Injectable()
+export class WebhookSecretGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request: Request = context.switchToHttp().getRequest();
+    const signature = request.headers['x-hub-signature-256'] as string;
+    const secret = Env.SECRET.WEBHOOK;
```

Comment:
PullProofAI

Consider adding null checks for `request.headers['x-hub-signature-256']` and `Env.SECRET.WEBHOOK` to prevent potential runtime errors.
```diff
+     if (!request.headers['x-hub-signature-256'] || !Env.SECRET.WEBHOOK) {
+       throw new UnauthorizedException('Missing signature or webhook secret');
+     }
```



<!-- This is an auto-generated comment by PullProofAI -->